FeedStoreChallenge/CoreDataFeedStore.swift

```diff
@@ -0,0 +1,136 @@
+//
+//  Created by Nazar Alwi on 27/05/21
+//  Copyright © 2021 Essential Developer. All rights reserved.
+//
+
+import CoreData
+
+public final class CoreDataFeedStore: FeedStore {
+    private let container: NSPersistentContainer
+    private let context: NSManagedObjectContext
+    
+    public init(storeURL: URL, bundle: Bundle = .main) throws {
+        container = try NSPersistentContainer.load(modelName: "FeedStore", url: storeURL, in: bundle)
+        context = container.newBackgroundContext()
+    }
+    
+    public func deleteCachedFeed(completion: @escaping DeletionCompletion) {
+        perform { context in
+            do {
+                try ManagedCache.find(in: context).map(context.delete).map(context.save)
+                completion(nil)
+            } catch {
+                completion(error)
```

Comment:
If there's a deletion error, the changes in the managed model aren't automatically reverted. They stay in memory.
To roll back the changes `context.rollback()` should be called

```suggestion
                context.rollback()
                completion(error)
```